File: frontend/src/elo.ts
const expectation = (r1, r2) => 1 / (1 + Math.pow(10, (r1 - r2) / 400));
const updateScore = (rA, K, eA, newA) => rA + K*(newA  - eA);

export const updateRankings = (rA: number, rB: number, selected: string) => {
    const eA = expectation(rB, rA);
    const eB = expectation(rA, rB);

    return {
        rA: updateScore(rA, 42, eA, selected === "A" ? 1 : 0),
        rB: updateScore(rB, 42, eB, selected === "B" ? 1 : 0)
    }
};

File: app/elo.py
def calc_expectation(r1: float, r2: float) -> float:
    return 1 / (1 + (10**(r1 - r2)) / 400)

def update_score(current: float, K: int, expectation: float, new: float) ->  float:
    return current + K*(new  - expectation)

def update_rankings(rA: float, rB: float, selected: str) -> dict:
    eA = calc_expectation(rB, rA)
    eB = calc_expectation(rA, rB)
    k_factor = 42

    return dict(
        rA=update_score(rA, k_factor, eA, 1 if selected == "A" else 0),
        rB=update_score(rB, k_factor, eB, 1 if selected == "B" else 0)
    )

